libmogwai-schedule/scheduler.c

```diff
@@ -866,7 +866,7 @@ entry_compare_cb (gconstpointer a_,
       g_debug ("%s: Comparing schedule entries ‘%s’ and ‘%s’ by peer priority: %d vs %d",
                G_STRFUNC, mws_schedule_entry_get_id (a),
                mws_schedule_entry_get_id (b), a_peer_priority, b_peer_priority);
-      return b_peer_priority - a_peer_priority;
+      return (b_peer_priority > a_peer_priority) ? 1 : -1;
```

Comment:
This won't produce stable sorting though, right? (I don't know how important it is though, so just calling it out in case you've missed it)

Reply:
We discussed this on Slack, and it turns out that GitHub chopped the `if (a_peer_priority != b_peer_priority)` out of the review context, so Joaquim missed it. That’s the bit which ensures the sort is stable, so there’s no issue here.

Reply:
As @pwithnall clarified on Slack, it's not affecting the stability because it accounts for the case where the priorities are equal earlier in the function (which is not shown in the PR). Thus: :+1: